common/error.hpp

```diff
@@ -13,31 +13,42 @@
 //    limitations under the License.
 
 #include <string>
+#include <system_error>
 #include <type_traits>
 
 #ifndef MENDER_COMMON_ERROR_HPP
 #define MENDER_COMMON_ERROR_HPP
 
 namespace mender::common::error {
 
-template <typename ErrorCodeType>
+template <typename ErrorType>
 class Error {
 public:
-	static_assert(
-		std::is_enum<ErrorCodeType>::value, "Error requires an enum type for error codes");
-	ErrorCodeType error_code;
+	ErrorType error;
```

Comment:
I changed the name to `error` from `error_code`, since now technically any error can be used. I guess `error_condition` is sort-of an error code, but it's a class, neither an enum nor int, so I thought this was more generic. Open to criticism though.

Reply:
Why not just `code`, I mean, we are already typing:
`.error()` as the prefix a fair few places, so less stuttering?

Reply:
Ok, I'll change it to `code` then. I also like that it's shorter, since we will be typing it a lot, I think.